### ISSUE TITLE
Serious performance issues with production application

### ISSUE THREAD

**Issue by drenfr01** [drenfr01/unionCapital#237]
Investigation so far has revealed several possible causes:

1) The Modulus DB is acting strangely. it's not charging us for storage and the analytics don't register any activity
2) The insertMemberData method, called by the All Members admin template, is horribly written and is a massive bottleneck. 
3) There are no indices on any of our collections.
4) when logging in, a user is registered for all collections at once. This causes a significant delay, particularly on slow connections, and so we should use template level subscriptions
5) We don't have oplog tailing configured on production.

**Comment by drenfr01** [comment]
1) Open support ticket with Modulus
2) disabled insertMemberData method, in progress re-writing insertMemberData
3) added indices
4) no work yet
5) will configure this on sunday

**Comment by drenfr01** [comment]
couple of there solutions

6) use meteorhacks:aggregate to take advantage of mongo groups
7) figure out why sometimes we wait on 11 methods or more (see below screenshot)

**Comment by drenfr01** [comment]
![screen shot 2015-07-10 at 5 33 37 pm](https://cloud.githubusercontent.com/assets/508707/8629497/d4d74342-2729-11e5-991c-8e4631779117.png)

**Comment by drenfr01** [comment]
7) in general reduce the number of things published down to server

**Comment by drenfr01** [comment]
So we can't do oplog tailing on Modulus without paying for dedicated DBs, because it requires replication set. See below for pricing:

I’m glad you’re interested in our services! You will need a dedicated instance of Mongo for oplog tailing. Our dedicated Mongo instances are competitively priced and are fully set up and managed by Modulus. You will be given access to the server itself, so you can set up Mongo with whatever options you like or need. Below is a pricing list of the server types/sizes available, which can be sharded or replicated in any combination.

Keep in mind pricing is per node, per month. When estimating pricing for a multi-node cluster, add the prices together. Ex: a 2 node High CPU Large replica set is 135×2 or $270 per month.

SSD, High CPU Large 
2 Cores (7x), 3.75GB mem, 28GB storage
$135/month

SSD, High Memory Large
2 Cores (6.5x), 7.5GB mem, 28GB storage
$210/month

SSD, High CPU X-Large 
4 Cores (14x), 7.5GB mem, 76GB storage
$250/month

SSD X-Large
4 Cores (13x), 15GB mem, 76GB storage
$420/month

**Comment by drenfr01** [comment]
in v2.2. we've 
fixed horrible method, resized photos, added indices. 

in v2.3 we'll do template level subscriptions, etc.

**Comment by drenfr01** [comment]
the only thing left is template level subscriptions. creating a separate ticket for that and closing this one
